src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java

```diff
@@ -91,46 +90,43 @@
     @Argument(fullName="doNotFixOverhangs", shortName="doNotFixOverhangs", doc="do not have the walker hard-clip overhanging sections of the reads", optional=true)
     protected boolean doNotFixOverhangs = false;
 
-    /**
-     * This stores all of the already-split reads and manages any processing (e.g. clipping overhangs) that happens to them.
-     * It will emit reads to the underlying writer as needed so we don't need to worry about any of that in this class.
-     */
-    protected OverhangFixingManager overhangManager;
+    private SAMFileGATKReadWriter outputWriter;
+    private OverhangFixingManager overhangManager;
+    ReadTransformer rnaReadTransform;
 
     @Override
-    protected Object doWork() {
-        IOUtil.assertFileIsReadable(INPUT);
-        final SamReader in = SamReaderFactory.makeDefault().open(INPUT);
-        final Iterable<GATKRead> readIter = new SAMRecordToReadIterator(in.iterator());
-        final SAMFileGATKReadWriter outputWriter = initialize(in);
-
-        final ReadTransformer rnaReadTransform = REFACTOR_NDN_CIGAR_READS ? new NDNCigarReadTransformer() : ReadTransformer.identity();
-
-        StreamSupport.stream(readIter.spliterator(), false)
-                .map(rnaReadTransform)
-                .forEach(read -> splitNCigarRead(read, overhangManager));
-        overhangManager.close();
-        CloserUtil.close(in);
-        CloserUtil.close(outputWriter);
-        return null;
+    public CountingReadFilter makeReadFilter() {
+        return new CountingReadFilter("ALL_READS", ReadFilterLibrary.ALLOW_ALL_READS);
     }
-
-    private SAMFileGATKReadWriter initialize(final SamReader in) {
-        final SAMFileHeader outputHeader = ReadUtils.cloneSAMFileHeader(in.getFileHeader());
-        //preSorted is set to false for this writer since ReadCoordinateComparator doesn't match HTSJDK's when an unmapped read has a mapped mate.
-        //Also, we don't want to assume that reads will be written in order by the manager because in deep, deep pileups it won't work
-        final SAMFileGATKReadWriter outputWriter = new SAMFileGATKReadWriter(ReadUtils.createCommonSAMWriter(OUTPUT, REFERENCE_SEQUENCE, outputHeader, false, false, false));
-
+    
+    @Override
+    public void onTraversalStart() {
+        final IndexedFastaSequenceFile referenceReader;
```

Comment:
I think this should probably be moved to be a class field and then be closed in the tool shutdown.